bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoDimmerHandler.java

```diff
@@ -536,37 +555,19 @@ protected void updateWemoState() {
         updateStatus(ThingStatus.ONLINE);
     }
 
-    public String getWemoURL(String actionService) {
-        URL descriptorURL = service.getDescriptorURL(this);
-        String wemoURL = null;
-        if (descriptorURL != null) {
-            String deviceURL = StringUtils.substringBefore(descriptorURL.toString(), "/setup.xml");
-            wemoURL = deviceURL + "/upnp/control/" + actionService + "1";
-            return wemoURL;
-        }
-        return null;
-    }
-
-    @SuppressWarnings("null")
-    public State getDateTimeState(String attributeValue) {
-        if (attributeValue != null) {
-            long value = 0;
-            try {
-                value = Long.parseLong(attributeValue) * 1000; // convert s to ms
-            } catch (NumberFormatException e) {
-                logger.warn("Unable to parse attributeValue '{}' for device '{}'; expected long", attributeValue,
-                        getThing().getUID());
-                return null;
-            }
-            ZonedDateTime zoned = ZonedDateTime.ofInstant(Instant.ofEpochMilli(value),
-                    TimeZone.getDefault().toZoneId());
-            State dateTimeState = new DateTimeType(zoned);
-            if (dateTimeState != null) {
-                logger.trace("New attribute '{}' received", dateTimeState);
-                return dateTimeState;
-            }
+    public @Nullable State getDateTimeState(String attributeValue) {
+        long value = 0;
+        try {
+            value = Long.parseLong(attributeValue) * 1000; // convert s to ms
```

Comment:
You can also use `Instant.ofEpochSecond` so you don't need this conversion.

Reply:
Done